File: frontend/lib/ai-chat.ts
import { apiFetch } from "@/lib/api";

export interface Conversation {
  id: string;
  title: string | null;
  task_id: string | null;
  created_at: string;
}

export interface ResearchSource {
  title: string;
  url: string;
  content: string;
}

export interface ChatMessage {
  id: string;
  role: "user" | "assistant";
  content: string;
  sources: ResearchSource[] | null;
  created_at: string;
}

export interface ChatResponse {
  conversation_id: string;
  message: ChatMessage;
}

export function listConversations() {
  return apiFetch<Conversation[]>("/api/v1/ai_chat/conversations");
}

export function listMessages(conversationId: string) {
  return apiFetch<ChatMessage[]>(`/api/v1/ai_chat/conversations/${conversationId}/messages`);
}

export function sendChatMessage(input: {
  conversation_id?: string | null;
  message: string;
  task_id?: string | null;
  use_research?: boolean;
}) {
  return apiFetch<ChatResponse>("/api/v1/ai_chat/chat", {
    method: "POST",
    body: JSON.stringify(input),
  });
}

File: backend/app/schemas/ai_chat.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, ConfigDict


class ConversationRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    title: str | None
    task_id: uuid.UUID | None
    created_at: datetime


class MessageRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    role: str
    content: str
    sources: Any | None
    created_at: datetime


class ChatRequest(BaseModel):
    conversation_id: uuid.UUID | None = None
    message: str
    task_id: uuid.UUID | None = None


class ChatResponse(BaseModel):
    conversation_id: uuid.UUID
    message: MessageRead
